catalog/core/catalog-core-subscriptionstore/subscriptionstore-impl/src/main/java/org/codice/ddf/catalog/subscriptionstore/common/CachedSubscription.java

```diff
@@ -0,0 +1,195 @@
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.subscriptionstore.common;
+
+import static java.lang.String.format;
+import static org.apache.commons.lang.Validate.notNull;
+
+import java.util.Dictionary;
+import java.util.Hashtable;
+import java.util.Optional;
+
+import org.codice.ddf.catalog.subscriptionstore.internal.SubscriptionFactory;
+import org.codice.ddf.catalog.subscriptionstore.internal.SubscriptionRegistrationException;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.ServiceRegistration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.event.Subscription;
+
+/**
+ * A data object for storing subscription information in a cache.
+ * <p>
+ * The data stored is a mix of persisted and locally cached information. This includes the
+ * original {@link Subscription} either passed from the client or deserialized from stored data,
+ * and the relevant {@link SubscriptionMetadata}.
+ * <p>
+ * The existence of a {@link CachedSubscription} object <i>between</i> subscription CRUD operations implies
+ * that:
+ * <ol>
+ * <li>
+ * The {@code SubscriptionMetadata} in this object is also in the {@code PersistentStore}.
+ * </li>
+ * <li>
+ * The {@code Subscription} in this object is registered as a service with OSGi <b>only if</b>
+ * {@link #isNotRegistered()} is false.
+ * </li>
+ * </ol>
+ * {@code CachedSubscription} objects are not meant to be reused. They have a 1-to-1 correspondence with
+ * a {@code SubscriptionMetadata} object in the backing store and are used to track that subscription's
+ * lifecycle in OSGi.
+ * <p>
+ * Given the dynamic nature of Karaf and the need to handle {@link javax.cache.event.CacheEntryEvent}s,
+ * there may be a brief period during system startup where cached subscriptions exist in the cache but
+ * are not registered. This problem should rectify itself as additional {@link SubscriptionFactory}
+ * instances are registered by subscription endpoints.
+ * <p>
+ * Currently, no guarantees are made regarding the state of this object <i>during</i> a subscription CRUD
+ * operation. Only one container operation may be active at a time.
+ *
+ * @see org.codice.ddf.catalog.subscriptionstore.SubscriptionContainerImpl
+ */
+public class CachedSubscription {
+    private static final Logger LOGGER = LoggerFactory.getLogger(CachedSubscription.class);
+
+    private static final String SUBSCRIPTION_ID_OSGI = "subscription-id";
+
+    private static final String EVENT_ENDPOINT = "event-endpoint";
+
+    private static final Integer INITIAL_CAPACITY = 4;
+
+    private final SubscriptionMetadata metadata;
+
+    private transient Subscription subscription;
```

Comment:
Good point. Since we're caching by reference only on the heap, serializable was not necessary. But depending on the caching provider it may become necessary. I'll make a note to patch and test this on the next PR.